src/docs/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -6,77 +6,291 @@ description: Learn how to add a single Flutter screen to your existing iOS app.
 - (BOOL)application:(UIApplication *)application
     didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"io.flutter" project:nil];
-  [self.flutterEngine runWithEntrypoint:nil];
+  self.flutterEngine = [[FlutterEngine alloc] initWithName:@"my flutter engine"];
+  // Runs the default Dart entrypoint with a default Flutter route.
+  [self.flutterEngine run];
   [GeneratedPluginRegistrant registerWithRegistry:self.flutterEngine];
   return [super application:application didFinishLaunchingWithOptions:launchOptions];
 }
 
 @end
 ```
 
-If you are writing in Swift, you can do the following in your `AppDelegate.swift`:
+</div>
+
+<div class="tab-pane" id="engine-swift-tab" role="tabpanel" aria-labelledby="engine-swift-tab" markdown="1">
 
+**In `AppDelegate.swift`:**
+
+<?code-excerpt "AppDelegate.swift" title?>
 ```swift
 import UIKit
 import Flutter
-import FlutterPluginRegistrant // Only if you have Flutter Plugins.
+import FlutterPluginRegistrant // Used to connect plugins.
 
 @UIApplicationMain
-class AppDelegate: FlutterAppDelegate {
-  var flutterEngine : FlutterEngine?;
-  // Only if you have Flutter plugins.
+class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate below.
+  lazy var flutterEngine = FlutterEngine(name: "my flutter engine")
+
   override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-    self.flutterEngine = FlutterEngine(name: "io.flutter", project: nil);
-    self.flutterEngine?.run(withEntrypoint: nil);
+    // Runs the default Dart entrypoint with a default Flutter route.
+    flutterEngine.run();
     GeneratedPluginRegistrant.register(with: self.flutterEngine);
     return super.application(application, didFinishLaunchingWithOptions: launchOptions);
   }
+}
+```
+
+</div>
+
+</div>{% comment %} FlutterEngine language tab end {% endcomment -%}
 
+### Show a FlutterViewController with your FlutterEngine
+
+The following example shows a generic ViewController with a UIButton hooked to
+present a [`FlutterViewController`]({{site.api}}/objcdoc/Classes/FlutterViewController.html).
+The `FlutterViewController` uses the `FlutterEngine` instance created in the
+`AppDelegate`.
+
+<ul class="nav nav-tabs" id="vc-language" role="tablist">
+  <li class="nav-item">
+    <a class="nav-link active" id="vc-objc" href="#vc-objc-tab" role="tab" aria-controls="vc-objc" aria-selected="true">Objective-C</a>
+  </li>
+  <li class="nav-item">
+    <a class="nav-link" id="vc-swift" href="#vc-swift-tab" role="tab" aria-controls="vc-swift" aria-selected="false">Swift</a>
+  </li>
+</ul>
+
+<div class="tab-content"> {% comment %} FlutterViewController language tab start {% endcomment -%}
+
+<div class="tab-pane active" id="vc-objc-tab" role="tabpanel" aria-labelledby="vc-objc-tab" markdown="1">
+
+<?code-excerpt "ViewController.m" title?>
+```objectivec
```

Comment:
I thought the markdown language marker was objective-c?

Reply:
I'm not sure which syntax highlighter library we're using for the website. `objective-c` and `objc` don't work.